site/content/integrate/apps/structure/_index.md

```diff
@@ -9,54 +9,14 @@ aliases:
-  in a programming language like JavaScript, Python, or Go.
+2. [Call]({{< ref "call" >}}) handlers, which are functions that handle incoming requests from the Mattermost server.
 
-- Static assets, currently solely icon files
+3. [Static assets]({{<ref "static-assets">}}), such as images.
 
-## Interactivity: Bindings and forms
+To interact with users, an App must [bind]({{< ref "bindings" >}}) a call handler to a [location]({{<ref "manifest#locations">}}) in the Mattermost user interface. These locations may be slash commands, toolbar and menu items, or embedded in posts.
 
-- To interact with users an App must [bind]({{< ref "bindings" >}}) a call to a
-  location in the Mattermost user interface. These locations may be
-  `/`-commands, toolbar and menu items, and embedded in posts.
+![image](apps-calls_bindings_locations_v2.svg)
 
-- A [binding]({{< ref "bindings" >}}) may display a [Form]({{< ref
-  "interactivity" >}}), or it may invoke a `Call` that will return a `Form`.
-  Forms allow to gather fields with limited dynamic behavior
-  (dynamically-populated selects, form refresh on field changes). A Form invokes
-  a `Call` when it is submitted. A `/`-command is just another way of filling
-  out a `Form` and submitting it.
-
-## Use Mattermost APIs
-
-Apps can use all general [Mattermost REST API]({{< ref "mattermost-api" >}})s,
-as well as special services that are provided to the apps.
-
-See [Authenticating with Mattermost]({{< ref "app-to-mattermost" >}}) for how
-to authenticate to all these services.
-
-App services:
-
-- [KV Store]({{< ref "mattermost-api#apps-api" >}}) - store and retrive app-specific data.
-
-- [OAuth2 Store]({{< ref "mattermost-api#apps-api" >}}) - store, expand, and retrieve user and
-  app OAuth2 configuration data.
-
-- [Calling other Apps]({{< ref "mattermost-api#apps-api" >}}) - use the `Call` API to invoke
-  other apps (experimental).
-
-## Other
-
-- [Subscriptions]({{< ref "subscriptions" >}}) covers how to subscribe to
-  notifications abdout Mattermost events. Examples of a subscription include having your App notified whenever a message is posted in a channel that mentions your @bot account.
-
-- [External webhooks]({{< ref "/integrate/apps/functionality/external-webhooks" >}}) covers subscribing to
-  webhooks from third-party systems.
-
-- [Lifecycle]({{< ref "lifecycle" >}}) covers lifecycle callback API, i. e.
-  `on_install`. This is useful for calling your app when it is first installed, or the user is attempting to uninstall it and gives you a chance to initialize or clean up data (such as from the KV store) before uninstalling your app.
-  
+A [Binding]({{< ref "bindings" >}}) may display a [Form]({{< ref "interactivity" >}}), or it may invoke a call that will return a form. Forms allow the user to input information with limited dynamic behavior, such as dynamically-populated pick lists, or form refreshes on field changes.
```

Comment:
```suggestion
A [Binding]({{< ref "bindings" >}}) may display a [Form]({{< ref "interactivity" >}}), or it may invoke a call to do something. The call may also return a form. Forms allow the user to input information with limited dynamic behavior, such as dynamically-populated pick lists, or form refreshes on field changes. Forms are submitted to call handlers.
```

Reply:
I think Forms as a topic needs to be a peer to Bindings since it seems to be vitally important to the Apps framework. I thought the "interactivity" of an App was Forms, however that's not quite the case; it's how data gets into an App.

I'll create a new card on my board to update the documentation once this PR merges so we can get a comprehensive explanation of Forms out there. :slightly_smiling_face: